src/Grpc.Net.ClientFactory/Internal/DefaultGrpcClientFactory.cs

```diff
@@ -53,19 +51,26 @@ internal class DefaultGrpcClientFactory : GrpcClientFactory
                 throw new InvalidOperationException($"No gRPC client configured with name '{name}'.");
             }
 
-            var httpClientFactoryOptions = _httpClientFactoryOptionsMonitor.Get(name);
-            if (httpClientFactoryOptions.HttpClientActions.Count > 0)
-            {
-                throw new InvalidOperationException($"The ConfigureHttpClient method is not supported when creating gRPC clients. Unable to create client with name '{name}'.");
-            }
+            var callInvoker = _callInvokerFactory.CreateInvoker(name, typeof(TClient));
 
             var clientFactoryOptions = _grpcClientFactoryOptionsMonitor.Get(name);
-            var httpHandler = _messageHandlerFactory.CreateHandler(name);
-            var callInvoker = _callInvokerFactory.CreateCallInvoker(httpHandler, name, typeof(TClient), clientFactoryOptions);
+
+            var resolvedCallInvoker = GrpcClientFactoryOptions.BuildInterceptors(
+                callInvoker,
+                _serviceProvider,
+                clientFactoryOptions,
+                InterceptorLifetime.Client);
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            if (clientFactoryOptions.Interceptors.Count != 0)
```

Comment:
Is there a universe where we need to support a mix of interceptors registered on the obsolete `Interceptors` list and the new `InterceptorRegistrations` list?

Reply:
No. The only way the old `Interceptors` collection could have interceptors is if someone manually added them (i.e. didn't use the interceptor extension methods).

This is for backward compatibility until the old collection gets removed.